pkg/core/operation.go

```diff
@@ -74,6 +74,23 @@ func (op *Operation) IsTokenOperation() bool {
 	return op.Type == OpTypeTokenActivatePool || op.Type == OpTypeTokenApproval || op.Type == OpTypeTokenCreatePool || op.Type == OpTypeTokenTransfer
 }
 
+func (op *Operation) DeepCopy() *Operation {
+	return &Operation{
+		ID:          op.ID,
+		Namespace:   op.Namespace,
+		Transaction: op.Transaction,
+		Type:        op.Type,
+		Status:      op.Status,
+		Plugin:      op.Plugin,
+		Input:       op.Input,
+		Output:      op.Output,
+		Error:       op.Error,
+		Created:     op.Created,
+		Updated:     op.Updated,
+		Retry:       op.Retry,
+	}
+}
```

Comment:
This DeepCopy needs to handle the fact that some of the fields are pointers so that they are deep copied correctly https://github.com/hyperledger/firefly/blob/74e2cbd38e688d0943618f149068880150a67020/pkg/core/operation.go#L114-L125 so needs a more sophisticated check for `nil` fields as well

Reply:
It works in your test because UUID is actually special in the case where it has no function to modify the value itself and in the test you are doing a complete replace